open_humans/views.py

```diff
@@ -93,13 +94,13 @@ class OAuth2LoginView(LargePanelMixin, TemplateView):
     template_name = 'account/login-oauth2.html'
 
     def get_context_data(self, **kwargs):
-        next_querystring = querydict_from_dict({
-            'next': self.request.GET.get('next')
-        }).urlencode()
 
+        # We need to retrieve the connection name out of the stored url to
+        # display it on the oauth2 login page.
+        redirect_url = self.request.session.get('next_url')
+        connection = parse_qs(urlparse(redirect_url).query).get('connection')[0]
```

Comment:
The `next_url` in session should only have the URL we want to redirect to – not extra stuff riding along it. That was the simplification, right? Then we aren't worried about parsing parameters. See above regarding retaining the current `connection` parameter behavior – only change `next`.